Firmware/eeprom.cpp

```diff
@@ -182,6 +136,20 @@ void eeprom_switch_to_next_sheet()
     if (sheet >= 0) eeprom_update_byte(&(EEPROM_Sheets_base->active_sheet), sheet);
 }
 
+bool __attribute__((noinline)) eeprom_is_sheet_initialized(uint8_t sheet_num) {
+    return (eeprom_read_word(reinterpret_cast<uint16_t*>(&(EEPROM_Sheets_base->s[sheet_num].z_offset))) != EEPROM_EMPTY_VALUE16);
+}
+
+bool __attribute__((noinline)) eeprom_is_initialized_block(const void *__p, size_t __n) {
+    const uint8_t *top = (const uint8_t*)__p + __n;
+    for (const uint8_t *p = (const uint8_t *)__p; p < top; p++) {
```

Comment:
iterating 16bit variables will take more space (a pointer is a 16bit variable)

Reply:
I've changed this in the meantime. I'm now using __n as a downcounter and incrementing __p